theano/tensor/nnet/spatialtf.py

```diff
@@ -0,0 +1,186 @@
+"""
+Abstract spatial transformer interface
+"""
+from __future__ import absolute_import, print_function, division
+
+import theano
+import theano.tensor as T
+from theano import Op
+from theano.tensor import as_tensor_variable
+from theano.gof import Apply
+from theano.gof.opt import local_optimizer
+from theano.tensor.opt import register_canonicalize
+
+class AbstractSpatialTransformerOp(Op):
+    def __init__(self, border_mode):
+        self.border_mode = border_mode
+
+    def make_node(self, inp, theta, scale_height, scale_width):
```

Comment:
I suggest that Abstract op here receives real width and height instead of scale_width and scale_height, as we do on GPU. This will give more options (e.g. by allowing to instanciate Abstract op with already known dimensions without having to use scales).

As on GPU, `spatialtf()` below should be the only one responsible for inferring real width and height before passing it to abstract op.